synapse/rest/client/devices.py

```diff
@@ -347,14 +356,55 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         return 200, result
 
 
+class DehydratedDeviceEventsServlet(RestServlet):
```

Comment:
Is it possible for multiple clients to be hitting this endpoint at once? What solves that?

Reply:
This would be something like a scenario where (let's say) two devices have called `GET /dehydrated_device` and received the dehydrated device id, and are now both calling this endpoint with the id? 
I don't think there is anything currently preventing this - iirc from the MSC proposal, this sort of scenario was the rationale for deleting the dehydrated device the first time this endpoint is called, as that functions as "claiming" the dehydrated device so it can't be used by other clients. This behavior was contentious on the MSC though - I implemented deleting the device after all the messages are delivered, but that won't help in the case where two devices are calling this endpoint at the same time.

Reply:
Yeah probably a conversation for the MSC, but seems quite likely now.